pontoon/base/views.py

```diff
@@ -582,6 +614,18 @@ def update_translation(request):
     user = request.user
     project = e.resource.project
 
+    # Read-only translations cannot saved
+    if (
+        ProjectLocale.objects.filter(
+            locale=locale,
+            project=project,
+            readonly=True,
+        ).exists()
+    ):
```

Comment:
I would put this in a function, to avoid repeating it so much. 

```python
if is_readonly(locale, project):
    return HttpResponseForbidden(
        "Forbidden: This string is in read-only mode"
    )
```

Or, even better, could that be a decorator? That would remove most of the boilerplate and make it a lot easier to use on all those views that edit translations. 

```python
def check_readonly_projetlocale(fn):
    def wrapper(request, *args, **kwargs):
        locale = request.get('locale')
        project = request.get('project')
        
        if (
            ProjectLocale.objects.filter(
                locale=locale,
                project=project,
                readonly=True,
            ).exists()
        ):
            return HttpResponseForbidden(
                "Forbidden: This string is in read-only mode"
            )

        return fn(request, *args, **kwargs)

    return wrapper
```

Wow I haven't written a decorator in a long while, not sure that's correct, haven't tested it... :stuck_out_tongue:

Reply:
Actually, given that you use the query in a bunch of different places, I'd combine the two and have a function that is called by the decorator.